library/src/main/java/com/pokegoapi/util/BosslandHash.java

```diff
@@ -0,0 +1,125 @@
+package com.pokegoapi.util;
+
+import okhttp3.*;
+import com.pokegoapi.api.PokemonGo;
+import lombok.Getter;
+import lombok.Setter;
+
+import java.io.IOException;
+import java.lang.IllegalAccessException;
+import java.lang.IllegalStateException;
+import java.lang.NullPointerException;
+import java.lang.System;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Base64;
+
+public class BosslandHash {
+    private static final String BASE = "http://pokehash.buddyauth.com/";
+    private static final String VERSION_END = "api/hash/versions";
+
+    private static String version;
+
+    public static class HashResponseContent {
+        @Getter private long locationAuthHash;
+        @Getter private long locationHash;
+        @Getter private List<Long> requestHashes;
+
+        public HashResponseContent(String response) {
+            response = response.replaceAll("\\D+", "|").substring(1);
+            String[] split = response.split("\\|");
+
+            locationAuthHash = Long.parseLong(split[0]);
+            locationHash = Long.parseLong(split[1]);
+
+            requestHashes = new ArrayList<>();
+            for (int i = 2; i < split.length; i++)
+                requestHashes.add(Long.parseLong(split[i]));
+        }
+
+    }
+
+    public static class HashRequestContent {
+        @Setter private long timestamp;
+        @Setter private double latitude;
+        @Setter private double longitude;
+        @Setter private double altitude;
+        @Setter private byte[] authTicket;
+        @Setter private byte[] sessionData;
+        @Getter private List<byte[]> requests;
+
+        public HashRequestContent() {
+            requests = new ArrayList<>();
+        }
+
+        public String json() {
+            Base64.Encoder encoder = Base64.getEncoder();
+            String auth = encoder.encodeToString(authTicket);
+            String session = encoder.encodeToString(sessionData);
+            String request = "";
+            for (byte[] array : requests)
+                request += "\"" + encoder.encodeToString(array) + "\",";
+            request = request.substring(0, request.length() - 1);
+
+            return "{"
```

Comment:
Didn't have the internet capacity to check out the imported libraries and how they work, please do add it if you can :)